### PR TITLE
Apple signin

### PR DESCRIPTION
**Pull request by renkelvin**
1. Add Apple sign in support
2. Add built-in OAuth provider (Twitter, Github etc.) methods
3. Add icon resources as part of the pod

### CODE REVIEW

File: OAuth/FirebaseOAuthUI/FUIOAuth.m [hunk 1]
```diff
@@ -117,12 +121,82 @@ - (instancetype)initWithAuthUI:(FUIAuth *)authUI
...
+      _provider = [FIROAuthProvider providerWithProviderID:self.providerID];
+    }
   }
   return self;
 }
 
++ (FUIOAuth *)twitterAuthProvider {
+  return [[FUIOAuth alloc] initWithAuthUI:[FUIAuth defaultAuthUI]
+                               providerID:@"twitter.com"
+                          buttonLabelText:@"Sign in with Twitter"
```
Comment (morganchen12) [inline OAuth/FirebaseOAuthUI/FUIOAuth.m]: do these button label texts need to be localized?

File: OAuth/FirebaseOAuthUI/FUIOAuth.h [hunk 2]
```diff
@@ -61,6 +61,31 @@ NS_ASSUME_NONNULL_BEGIN
...
+
+/** @fn yahooAuthProvider
+    @brief Built-in OAuth provider for Yahoo.
+*/
++ (FUIOAuth *)yahooAuthProvider;
+
+/** @fn appleAuthProvider
+    @brief Built-in OAuth provider for Apple.
+*/
++ (FUIOAuth *)appleAuthProvider;
```
Comment (morganchen12) [inline OAuth/FirebaseOAuthUI/FUIOAuth.h]: nit: this method should be marked `API_AVAILABLE(ios(13.0))`

File: OAuth/FirebaseOAuthUI/FUIOAuth.m [hunk 3]
```diff
@@ -212,6 +300,27 @@ - (BOOL)handleOpenURL:(NSURL *)URL sourceApplication:(nullable NSString *)source
   return NO;
 }
 
+#pragma - ASAuthorizationControllerDelegate
```
Comment (morganchen12) [inline OAuth/FirebaseOAuthUI/FUIOAuth.m]: `#pragma mark`

File: OAuth/FirebaseOAuthUI/FUIOAuth.m [hunk 4]
```diff
@@ -212,6 +300,27 @@ - (BOOL)handleOpenURL:(NSURL *)URL sourceApplication:(nullable NSString *)source
...
+                                                                      IDToken:idToken
+                                                                  accessToken:nil];
+  _providerSignInCompletion(credential, nil, nil, nil);
+}
+
+- (void)authorizationController:(ASAuthorizationController *)controller didCompleteWithError:(NSError *)error API_AVAILABLE(ios(13.0)){
+  NSLog(@"%@", error.description);
+}
+
+#pragma - ASAuthorizationControllerPresentationContextProviding
```
Comment (morganchen12) [inline OAuth/FirebaseOAuthUI/FUIOAuth.m]: `#pragma mark`

File: OAuth/FirebaseOAuthUI/FUIOAuth.m [hunk 5]
```diff
@@ -212,6 +300,27 @@ - (BOOL)handleOpenURL:(NSURL *)URL sourceApplication:(nullable NSString *)source
...
+- (void)authorizationController:(ASAuthorizationController *)controller didCompleteWithAuthorization:(ASAuthorization *)authorization API_AVAILABLE(ios(13.0)){
+  ASAuthorizationAppleIDCredential* appleIDCredential = authorization.credential;
+  NSString *idToken = [NSString stringWithUTF8String:[appleIDCredential.identityToken bytes]];
+  FIROAuthCredential *credential = [FIROAuthProvider credentialWithProviderID:@"apple.com"
+                                                                      IDToken:idToken
+                                                                  accessToken:nil];
+  _providerSignInCompletion(credential, nil, nil, nil);
+}
+
+- (void)authorizationController:(ASAuthorizationController *)controller didCompleteWithError:(NSError *)error API_AVAILABLE(ios(13.0)){
```
Comment (morganchen12) [inline OAuth/FirebaseOAuthUI/FUIOAuth.m]: nit: space before `{` at end of line

File: OAuth/FirebaseOAuthUI/FUIOAuth.m (hunk 1 above)
Comment (renkelvin) [reply OAuth/FirebaseOAuthUI/FUIOAuth.m]: Seems we don't localize any strings in FirebaseUI currently. While I think it's a good idea.

**Pull request merged**